src/api/minecraft-api.js

```
@@ -366,119 +400,94 @@ class MinecraftApi {
-                console.log('MinecraftServer stopped.');
-                this.stopPollers();
-            });
-        }
-        console.log('MinecraftApi stopped.');
+        console.log(`Starting MinecraftServer...`);
+        await minecraftServer.start();
+        console.log(`MinecraftServer started.`);
+        this.pollMinecraft(100);
     }
 
-    stopPollers () {
+    /**
+     * Stops the MinecraftServer instance.
+     */
+    async stopMinecraft () {
         let properties = this.properties;
+        let minecraftServer = properties.minecraftServer;
 
-        if (properties.pollers.minecraftStatusTimerId) {
-            clearTimeout(properties.pollers.minecraftStatusTimerId);
-        }
+        this.stopMinecraftPoller();
+        await minecraftServer.stop();
+        return Promise.resolve();
     }
 
-    startMinecraft (callback) {
-        let properties = this.properties,
-            minecraftServer = properties.minecraftServer;
-
-        minecraftServer.start(() => {
-            this.startPollers(100);
-            if (typeof callback === 'function') {
-                callback();
-            }
-        });
+    /**
+     * Stops and starts the MinecraftServer instance.
+     */
+    async restartMinecraft () {
+        await this.stopMinecraft();
+        await this.startMinecraft();
+        return Promise.resolve();
     }
 
-    stopMinecraft (callback) {
-        let properties = this.properties,
-            minecraftServer = properties.minecraftServer;
-
-        this.stopPollers();
-        minecraftServer.stop(() => {
-            if (typeof callback === 'function') {
-                callback();
-            }
-        });
+    /**
+     * Logs things to a file.
+     * @param {string} data The data to log.
+     */
+    async log (data) {
+        try {
+            await Util.log(data, 'minecraft-api.log');
+            return Promise.resolve();
+        } catch (err) {
+            return Promise.reject(err);
```

Rather than explicitly returning a rejected `Promise`, `throw err;`